plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/HdfsOrcDataSource.java

```diff
@@ -34,26 +33,26 @@
 public class HdfsOrcDataSource
         extends AbstractOrcDataSource
 {
-    private final FSDataInputStream inputStream;
+    private final TrinoInput input;
     private final FileFormatDataSourceStats stats;
 
     public HdfsOrcDataSource(
             OrcDataSourceId id,
             long size,
             OrcReaderOptions options,
-            FSDataInputStream inputStream,
+            TrinoInput input,
```

Comment:
It would be cleaner if this was `TrinoInputFile`. Then `HdfsOrcDataSource` would really own lifecycle of `TrinoInput` (both create it and close it).